### PR TITLE
created user sign up, not receiving emails

### PR DESCRIPTION
**Pull request by dawnaara**
@cbeck I created the user sign up with devise. However, I am not receiving registration emails. i set up a mailer. Do I need to set up another route or model?

### CODE REVIEW

File: app/controllers/user_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class UserController < ApplicationController
+	before_action :authenticate_user!
+
+	def index
```
Comment (cbeck) [inline app/controllers/user_controller.rb]: You don't need this.

File: app/controllers/user_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	end
+
+	def show
+	    @user = User.find(params[:id])
+		unless @user == current_user
+        	redirect_to :back, :alert => "Access denied."	
+		end
+	end
+
+	def new
```
Comment (cbeck) [inline app/controllers/user_controller.rb]: Or this

File: app/controllers/user_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		@user = User.new
+	end
+
+	def edit
```
Comment (cbeck) [inline app/controllers/user_controller.rb]: Or this

File: app/controllers/user_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	end
+
+   def create
```
Comment (cbeck) [inline app/controllers/user_controller.rb]: Or this

File: app/mailers/application_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
```
Comment (cbeck) [inline app/mailers/application_mailer.rb:1]: You don't need this

File: app/mailers/user_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class UserMailer < ApplicationMailer
```
Comment (cbeck) [inline app/mailers/user_mailer.rb]: You don't need this

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
```
Comment (cbeck) [inline app/models/user.rb:3]: If you want to receive emails from devise for sign up, you need to take this :confirmable part and put it with the other plugins for devise (like after :validatable on the next line)

File: app/views/layouts/mailer.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<html>
```
Comment (cbeck) [inline app/views/layouts/mailer.html.erb]: You don't need this

File: app/views/layouts/mailer.text.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<%= yield %>
```
Comment (cbeck) [inline app/views/layouts/mailer.text.erb]: Or this

File: app/views/user/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,36 @@
+ <h2>Sign up</h2>
```
Comment (cbeck) [inline app/views/user/new.html.erb]: You don't need this

File: app/views/user_mailer/registration_confirmation.html.erb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<p>Hello <%= @user.name %>!</p>
```
Comment (cbeck) [inline app/views/user_mailer/registration_confirmation.html.erb]: You don't need this

File: app/views/user_mailer/registration_confirmation.text.erb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+UserMailer#registration_confirmation
```
Comment (cbeck) [inline app/views/user_mailer/registration_confirmation.text.erb]: You don't need this

File: config/routes.rb [hunk 13]
```diff
@@ -1,56 +1,5 @@
...
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+root 'user#index'
```
Comment (cbeck) [inline config/routes.rb]: You probably want to change this to use a welcome#index and create a welcome controller with an index method.

File: db/migrate/20160804035704_devise_create_users.rb [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.string   :current_sign_in_ip
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
```
Comment (cbeck) [inline db/migrate/20160804035704_devise_create_users.rb]: For confirmable to work, you'll need to uncomment lines 23 - 26

File: test/controllers/user_controller_test.rb [hunk 15]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (cbeck) [inline test/controllers/user_controller_test.rb]: You don't need this

File: test/mailers/previews/user_mailer_preview.rb [hunk 16]
```diff
@@ -0,0 +1,9 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
```
Comment (cbeck) [inline test/mailers/previews/user_mailer_preview.rb]: You don't need this

File: test/mailers/user_mailer_test.rb [hunk 17]
```diff
@@ -0,0 +1,12 @@
+require 'test_helper'
```
Comment (cbeck) [inline test/mailers/user_mailer_test.rb]: You don't need this

File: test/models/user_test.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (cbeck) [inline test/models/user_test.rb]: You don't need this

**Comment by cbeck** [comment]
Hi Dawna, 

I thought I gave you a heads-up about this, but devise does all of this user registration and login stuff for you. I made a bunch of comments in your code on things you need to remove - basically all of your methods inside your users controller except the show method, which you'll need later, your user tests, views, mailers, etc. That's all baked into devise. 

The reason why you were not getting any email is because in order to get that, devise needs the confirmable plugin activated, which I noted in your user model. Additionally, in the migration for users that devise generated, you will need to uncomment the confirmable fields. After you do, you'll want to:

```
rake db:drop
rake db:migrate
```

The last step in getting it to actually send the email is you need to setup figaro and use SendGrid, just like you did the favorite mailer stuff (but without having to create the actual mailer which is part of the devise gem). So I am talking about that setup_mail.rb file in initializers and the changes to the development.rb file.

Hope this helps!

**Comment by cbeck** [comment]
Btw - do NOT merge this PR as it sits.

**Comment by dawnaara** [comment]
Yeah, we did go over devise but when I wasn't getting the email I tried to
troubleshoot on my own. Guess I went down the wrong rabbit hole!

**Comment by dawnaara** [comment]
@cbeck couldn't figure out how to get pull request working so I went ahead and merged the branch. Hope it was ok to do so and I didn't mess anything up. We will see...

**Pull request merged**